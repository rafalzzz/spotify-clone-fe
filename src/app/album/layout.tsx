import { FC, PropsWithChildren } from 'react';

const Album: FC<PropsWithChildren> = ({ children }): JSX.Element => <>{children}</>;

export default Album;
