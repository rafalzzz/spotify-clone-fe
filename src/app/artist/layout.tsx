import { FC, PropsWithChildren } from 'react';

const Artist: FC<PropsWithChildren> = ({ children }) => <>{children}</>;

export default Artist;
