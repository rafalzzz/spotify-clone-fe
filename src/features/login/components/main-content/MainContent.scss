.main-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 2rem;
  background-color: #000;

  .reset-password-redirection {
    &__hyperlink {
      padding-left: 5px;
      color: var(--main-text-color);

      &:hover {
        color: var(--spotify-hyperlink-color);
      }
    }
  }
}
