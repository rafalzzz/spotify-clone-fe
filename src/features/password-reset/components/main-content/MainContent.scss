.main-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 2rem;
  background-color: #000;
}
