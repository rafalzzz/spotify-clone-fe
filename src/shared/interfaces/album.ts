export interface Album {
  artistId: number;
  artistName: string;
  artworkUrl60: string;
  artworkUrl100: string;
  collectionCensoredName: string;
  collectionExplicitness: string;
  collectionId: number;
  collectionName: string;
  collectionPrice: number;
  collectionType: string;
  collectionViewUrl: string;
  copyright: string;
  country: string;
  currency: string;
  primaryGenreName: string;
  releaseDate: string;
  trackCount: number;
  wrapperType: string;
}
