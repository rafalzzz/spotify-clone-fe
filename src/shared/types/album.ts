export const enum EAlbumKeys {
  ARTIST_ID = 'artistId',
  ARTIST_NAME = 'artistName',
  ARTWORK_URL_60 = 'artworkUrl60',
  ARTWORK_URL_100 = 'artworkUrl100',
  COLLECTION_CENSORED_NAME = 'collectionCensoredName',
  COLLECTION_EXPLICITNESS = 'collectionExplicitness',
  COLLECTION_ID = 'collectionId',
  COLLECTION_NAME = 'collectionName',
  COLLECTION_PRICE = 'collectionPrice',
  COLLECTION_TYPE = 'collectionType',
  COLLECTION_VIEW_URL = 'collectionViewUrl',
  COPYRIGHT = 'copyright',
  COUNTRY = 'country',
  CURRENCY = 'currency',
  PRIMARY_GENRE_NAME = 'primaryGenreName',
  RELEASE_DATE = 'releaseDate',
  TRACK_COUNT = 'trackCount',
  WRAPPER_TYPE = 'wrapperType',
}

export type Album = {
  [EAlbumKeys.ARTIST_ID]: number;
  [EAlbumKeys.ARTIST_NAME]: string;
  [EAlbumKeys.ARTWORK_URL_60]: string;
  [EAlbumKeys.ARTWORK_URL_100]: string;
  [EAlbumKeys.COLLECTION_CENSORED_NAME]: string;
  [EAlbumKeys.COLLECTION_EXPLICITNESS]: string;
  [EAlbumKeys.COLLECTION_ID]: number;
  [EAlbumKeys.COLLECTION_NAME]: string;
  [EAlbumKeys.COLLECTION_CENSORED_NAME]: string;
  [EAlbumKeys.COLLECTION_PRICE]: number;
  [EAlbumKeys.COLLECTION_TYPE]: string;
  [EAlbumKeys.COLLECTION_VIEW_URL]: string;
  [EAlbumKeys.COPYRIGHT]: string;
  [EAlbumKeys.COUNTRY]: string;
  [EAlbumKeys.CURRENCY]: string;
  [EAlbumKeys.PRIMARY_GENRE_NAME]: string;
  [EAlbumKeys.RELEASE_DATE]: string;
  [EAlbumKeys.TRACK_COUNT]: number;
  [EAlbumKeys.WRAPPER_TYPE]: string;
};
