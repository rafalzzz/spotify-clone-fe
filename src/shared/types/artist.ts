export const enum EArtistKeys {
  AMG_ARTIST_ID = 'amgArtistId',
  ARTIST_ID = 'artistId',
  ARTIST_LINK_URL = 'artistLinkUrl',
  ARTIST_NAME = 'artistName',
  ARTIST_TYPE = 'artistType',
  PRIMARY_GENRE_ID = 'primaryGenreId',
  PRIMARY_GENRE_NAME = 'primaryGenreName',
  WRAPPER_TYPE = 'wrapperType',
}

export type Artist = {
  [EArtistKeys.AMG_ARTIST_ID]: number;
  [EArtistKeys.ARTIST_ID]: number;
  [EArtistKeys.ARTIST_LINK_URL]: string;
  [EArtistKeys.ARTIST_NAME]: string;
  [EArtistKeys.ARTIST_TYPE]: string;
  [EArtistKeys.PRIMARY_GENRE_ID]: number;
  [EArtistKeys.PRIMARY_GENRE_NAME]: string;
  [EArtistKeys.WRAPPER_TYPE]: string;
};
