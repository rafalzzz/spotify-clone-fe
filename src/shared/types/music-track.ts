export const enum EMusicTrackKeys {
  ARTIST_ID = 'artistId',
  ARTIST_NAME = 'artistName',
  ARTWORK_URL_30 = 'artworkUrl30',
  ARTWORK_URL_60 = 'artworkUrl60',
  ARTWORK_URL_100 = 'artworkUrl100',
  COLLECTION_CENSORED_NAME = 'collectionCensoredName',
  COLLECTION_EXPLICITNESS = 'collectionExplicitness',
  COLLECTION_ID = 'collectionId',
  COLLECTION_NAME = 'collectionName',
  COLLECTION_PRICE = 'collectionPrice',
  COLLECTION_VIEW_URL = 'collectionViewUrl',
  COUNTRY = 'country',
  CURRENCY = 'currency',
  DISC_COUNT = 'discCount',
  DISC_NUMBER = 'discNumber',
  IS_STREAMABLE = 'isStreamable',
  KIND = 'kind',
  PREVIEW_URL = 'previewUrl',
  PRIMARY_GENRE_NAME = 'primaryGenreName',
  RELEASE_DATE = 'releaseDate',
  TRACK_CENSORED_NAME = 'trackCensoredName',
  TRACK_COUNT = 'trackCount',
  TRACK_EXPLICITNESS = 'trackExplicitness',
  TRACK_ID = 'trackId',
  TRACK_NAME = 'trackName',
  TRACK_NUMBER = 'trackNumber',
  TRACK_PRICE = 'trackPrice',
  TRACK_TIME_MILLIS = 'trackTimeMillis',
  TRACK_VIEW_URL = 'trackViewUrl',
  WRAPPER_TYPE = 'wrapperType',
}

export type TMusicTrack = {
  [EMusicTrackKeys.ARTIST_ID]: number;
  [EMusicTrackKeys.ARTIST_NAME]: string;
  [EMusicTrackKeys.ARTWORK_URL_30]: string;
  [EMusicTrackKeys.ARTWORK_URL_60]: string;
  [EMusicTrackKeys.ARTWORK_URL_100]: string;
  [EMusicTrackKeys.COLLECTION_CENSORED_NAME]: string;
  [EMusicTrackKeys.COLLECTION_EXPLICITNESS]: string;
  [EMusicTrackKeys.COLLECTION_ID]: number;
  [EMusicTrackKeys.COLLECTION_NAME]: string;
  [EMusicTrackKeys.COLLECTION_PRICE]: number;
  [EMusicTrackKeys.COLLECTION_VIEW_URL]: number;
  [EMusicTrackKeys.COUNTRY]: string;
  [EMusicTrackKeys.CURRENCY]: string;
  [EMusicTrackKeys.DISC_COUNT]: number;
  [EMusicTrackKeys.DISC_NUMBER]: number;
  [EMusicTrackKeys.IS_STREAMABLE]: boolean;
  [EMusicTrackKeys.KIND]: string;
  [EMusicTrackKeys.PREVIEW_URL]: string;
  [EMusicTrackKeys.PRIMARY_GENRE_NAME]: string;
  [EMusicTrackKeys.RELEASE_DATE]: string;
  [EMusicTrackKeys.TRACK_CENSORED_NAME]: string;
  [EMusicTrackKeys.TRACK_COUNT]: number;
  [EMusicTrackKeys.TRACK_EXPLICITNESS]: string;
  [EMusicTrackKeys.TRACK_ID]: number;
  [EMusicTrackKeys.TRACK_NAME]: string;
  [EMusicTrackKeys.TRACK_NUMBER]: number;
  [EMusicTrackKeys.TRACK_PRICE]: number;
  [EMusicTrackKeys.TRACK_TIME_MILLIS]: number;
  [EMusicTrackKeys.TRACK_VIEW_URL]: string;
  [EMusicTrackKeys.WRAPPER_TYPE]: string;
};
