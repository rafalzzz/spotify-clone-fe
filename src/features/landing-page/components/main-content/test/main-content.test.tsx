import { render, waitFor } from '@testing-library/react';

import { Album } from '@/shared/interfaces/album';
import { Artist } from '@/shared/interfaces/artist';
import { MusicTrack } from '@/shared/interfaces/music-track';

import MainContent from '..';

const props = {
  artists: [
    {
      artistId: 1,
      artistName: 'Test_Artist1',
    },
  ] as Artist[],
  albums: [
    {
      collectionId: 1,
      artworkUrl60: '/some-image-url1.jpg',
      collectionName: 'CollectionName',
      releaseDate: '2020-01-01',
      artistName: 'Test_Artist',
    },
  ] as Album[],
  songs: [
    {
      trackId: 1,
      artworkUrl60: '/some-image-url1.jpg',
      trackName: 'TestTrack',
      artistName: 'Test_Artist',
      collectionName: 'collectionName',
    },
  ] as MusicTrack[],
};

const renderLoveAlbumsSection = () => render(<MainContent {...props} />);

describe('MainContent', () => {
  it('render component without error', async () => {
    const { container } = renderLoveAlbumsSection();

    await waitFor(() => {
      expect(container).toBeDefined();
    });

    expect(container).toMatchSnapshot();
  });
});
