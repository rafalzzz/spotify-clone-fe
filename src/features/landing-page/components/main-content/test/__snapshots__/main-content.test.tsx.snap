// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent render component without error 1`] = `
<div>
  <main
    class="custom-content-wrapper"
  >
    <h1>
      Welcome
    </h1>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/love-songs"
          >
            <h2>
              Love songs
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/love-songs"
        >
          Show all
        </a>
      </div>
      <ul
        class="custom-section__items"
      >
        <li
          data-testid="section-item"
        >
          <a
            class="custom-section-item__redirection"
            data-testid="custom-section-item-redirection"
            href="/track/1"
          >
            <div
              class="ant-dropdown-trigger custom-section-item"
            >
              <div
                class="custom-section-item-image"
              >
                <img
                  alt="image"
                  data-nimg="1"
                  decoding="async"
                  height="100"
                  loading="lazy"
                  src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                  srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                  style="color: transparent; height: 100%; width: 100%;"
                  width="100"
                />
                <button
                  class="custom-section-item-play-button "
                  data-testid="custom-section-item-play-button"
                >
                  <span
                    aria-label="play-circle"
                    class="anticon anticon-play-circle antd-icon"
                    data-testid="custom-section-item-play-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="play-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="music-track-informations"
              >
                <div
                  data-mouseenterdelay="0.5"
                  data-placement="right"
                  data-testid="music-track-informations-track-name-tooltip"
                >
                  <h3
                    class="music-track-informations__text"
                    data-testid="music-track-informations-track-name"
                  >
                    TestTrack
                  </h3>
                  <span>
                    TestTrack
                  </span>
                </div>
                <button
                  class="music-track-informations__text music-track-informations__artist"
                  data-testid="music-track-informations-artist-name"
                >
                  <div
                    data-mouseenterdelay="0.5"
                    data-placement="right"
                    data-testid="music-track-informations-artist-name-tooltip"
                  >
                    <span>
                      Test_Artist
                    </span>
                    <span>
                      Test_Artist
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </section>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/love-albums"
          >
            <h2>
              Love albums
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/love-albums"
        >
          Show all
        </a>
      </div>
      <ul
        class="custom-section__items"
      >
        <li
          data-testid="section-item"
        >
          <a
            class="custom-section-item__redirection"
            data-testid="custom-section-item-redirection"
            href="/album/1"
          >
            <div
              class="ant-dropdown-trigger custom-section-item"
            >
              <div
                class="custom-section-item-image"
              >
                <img
                  alt="image"
                  data-nimg="1"
                  decoding="async"
                  height="100"
                  loading="lazy"
                  src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                  srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                  style="color: transparent; height: 100%; width: 100%;"
                  width="100"
                />
                <button
                  class="custom-section-item-play-button "
                  data-testid="custom-section-item-play-button"
                >
                  <span
                    aria-label="play-circle"
                    class="anticon anticon-play-circle antd-icon"
                    data-testid="custom-section-item-play-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="play-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="album-information"
              >
                <div
                  data-mouseenterdelay="0.5"
                  data-placement="right"
                  data-testid="album-information-collection-name-tooltip"
                >
                  <h3
                    class="album-information__text"
                    data-testid="album-information-collection-name"
                  >
                    CollectionName
                  </h3>
                  <span>
                    CollectionName
                  </span>
                </div>
                <div
                  class="album-information__wrapper"
                >
                  <span
                    class="album-information__release-date"
                    data-testid="album-information-release-date"
                  >
                    2020
                  </span>
                  <span
                    class="album-information__separator"
                  >
                    â€¢
                  </span>
                  <button
                    class="album-information__text album-information__artist"
                    data-testid="album-information-artist-name"
                  >
                    <div
                      data-mouseenterdelay="0.5"
                      data-placement="right"
                      data-testid="album-information-artist-name-tooltip"
                    >
                      <span>
                        Test_Artist
                      </span>
                      <span>
                        Test_Artist
                      </span>
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </section>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/most-famous-djs"
          >
            <h2>
              Most Popular DJs
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/most-famous-djs"
        >
          Show all
        </a>
      </div>
      <ul
        class="custom-section__items"
      >
        <li>
          <a
            class="ant-dropdown-trigger custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/1"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist1
              </div>
            </div>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
